2199/2/versus.rb

```diff
@@ -0,0 +1,138 @@
+require 'yaml'
+require 'bundler'
+Bundler.require
+
+# :reek:InstanceVariableAssumption
+# class Battle to collect information
+class Battle
+  def initialize(filename)
+    @filename = filename
+  end
+
+  def bad_words_count
+    words.count do |word|
+      word.include?('*') || RussianObscenity.obscene?(word)
+    end
+  end
+
+  def words_count
+    words.count
+  end
+
+  def rounds_count
+    rounds_count = text.scan(/Раунд \d/).length
+    rounds_count.zero? ? 1 : rounds_count
+  end
+
+  private
+
+  def text
+    @text ||= File.read(@filename)
+  end
+
+  def words
+    return @words if @words
+    @words = text.scan(/[а-яА-ЯёЁ*]+/)
+    @words.delete('***')
+    @words
+  end
+end
+
+# Participant container
+class Participant
+  attr_reader :name
+
+  def initialize(name)
+    @name = name
+    @battles = []
+  end
+
+  def add_battle(filename)
+    @battles << Battle.new(filename)
```

Comment:
When you call `Battle.new` in `Participant` class you kind of make `Participant` dependent on `Battle`. Imagine soon you'll need to work not only with battles but also with disses. Your future `Diss` class will have all the needed methods like `#bad_words_count`. And in order to make `Participant` work with `Diss` you'll need to change much code in `Participant`.

What I'm trying to say is you need to pass a battle instance to `#add_battle` method instead of passing `filename`.

Read more about it in [POODR](http://www.poodr.com/).
Also, read about dependency injection.